geowebcache/core/src/main/java/org/geowebcache/storage/BlobStoreAggregator.java

```diff
@@ -0,0 +1,260 @@
+        }
+
+        // Check whether there is any general service information
+        if (this.serviceInformation == null && config instanceof ServerConfiguration) {
+            log.debug("Reading service information.");
+            this.serviceInformation = ((ServerConfiguration) config).getServiceInformation();
+        }
+        return blobStoreCount;
+    }
+
+    public ServiceInformation getServiceInformation() {
+        return this.serviceInformation;
+    }
+
+    /**
+     * @see org.springframework.beans.factory.DisposableBean#destroy()
+     */
+    public void destroy() throws Exception {
+        //
+    }
+
+    /**
+     * Finds out which {@link BlobStoreConfiguration} contains the given blob store, removes it, and saves the configuration.
+     *
+     * @param blobStoreName the name of the blob store to remove
+     * @return true if the blob store was removed, false if it was not found.
+     */
+    public synchronized void removeBlobStore(final String blobStoreName)
+            throws IllegalArgumentException {
+        for (BlobStoreConfiguration config : configs) {
+            if (config.containsBlobStore(blobStoreName)) {
+                config.removeBlobStore(blobStoreName);
+                return;
+            }
+        }
+        throw new NoSuchElementException("No configuration found containing blob store "+blobStoreName);
+    }
+    /**
+     * Adds a blob store and returns the {@link BlobStoreConfiguration} to which the blob stores was added.
+     *
+     * @param bs the blob store to add
+     * @throws IllegalArgumentException if the given blob store can't be added to any configuration managed by this blob
+     *         store aggregator.
+     */
+    public synchronized void addBlobStore(final BlobStoreInfo bs) throws IllegalArgumentException {
+        for (BlobStoreConfiguration c : configs) {
+            if (c.canSave(bs)) {
+                c.addBlobStore(bs);
+                return;
+            }
+        }
+        throw new IllegalArgumentException("No configuration found capable of saving " + bs);
+    }
+
+    /**
+     * Renames an existing blob store.
+     *
+     * @param oldName The name of the existing blob store
+     * @param newName The name to rename the blob store to
+     * @throws IllegalArgumentException
+     */
+    public synchronized void rename(final String oldName, final String newName) throws NoSuchElementException, IllegalArgumentException {
+        BlobStoreConfiguration config = getConfiguration(oldName);
+        config.renameBlobStore(oldName, newName);
+    }
+
+    /**
+     * Replaces the given blob store and returns the Blob Store's configuration, saving the configuration.
+     *
+     * @param bs The blob store to modify
+     * @throws IllegalArgumentException
+     */
+    public synchronized void modify(final BlobStoreInfo bs) throws IllegalArgumentException {
```

Comment:
Reminder, lets try and keep the method names consistent between BlobStoreConfugration and BlobStoreAgregator; So in this case, `renameBlobStore` and `modifyBlobStore`